### PR TITLE
Shell profiles

### PR DESCRIPTION
**Pull request by gimenete**
- Adds the `--profile foo` global flag
- Allows setting the profile values with env variables like `XATA_API_PROVIDER` (comma separated of apiHost,workspacesHost) or `XATA_WEB_URL`
- Unify host resolution code

Example:

```
# Log in with multiple accounts to production
xata auth login --profile tudor

# Log in to staging
xata auth login --profile staging --host staging

# Log in to local docker
xata auth login --profile local --host http://localhost:6000/,http://{workspaceId}.localhost:6000

# Use a different profile per command
xata import csv --profile staging ...

# Log out (delete) from a given profile
xata auth logout --profile tudor
```

### CODE REVIEW

File: cli/src/commands/auth/login.ts [hunk 1]
```diff
@@ -1,37 +1,50 @@
...
+
+    const profile = await this.getProfile(true);
+    const loggedIn = await hasProfile(profile.name);
+    if (loggedIn) {
       const { overwrite } = await this.prompt(
         {
           type: 'confirm',
           name: 'overwrite',
-          message: 'Authentication is already configured, do you want to overwrite it?'
+          message: `Authentication is already configured for ${profile.name} profile, do you want to overwrite it?`
```
Comment (gimenete) [inline cli/src/commands/auth/login.ts:31]:
```suggestion
          message: `Authentication is already configured for the ${profile.name} profile, do you want to overwrite it?`
```

File: cli/src/commands/auth/login.ts [hunk 2]
```diff
@@ -1,37 +1,50 @@
...
         },
         flags.force
       );
       if (!overwrite) this.exit(2);
     }
 
+    const host = parseProviderString(flags.host);
+    if (!host) {
+      this.error('Invalid host provider, expected either "production", "staging" or "apiUrl,workspacesUrl"');
```
Comment (gimenete) [inline cli/src/commands/auth/login.ts:40]:
```suggestion
      this.error('Invalid host provider, expected either "production", "staging" or "{apiUrl},{workspacesUrl}"');
```

To make it clear that the example are placeholders.

**Review by gimenete — COMMENTED** [review]
Great job!

I left a couple of comments.

I cannot approve because I'm the original author 😅  but LGTM 👍

**Review by SferaDev — APPROVED** [review]

File: cli/src/commands/auth/login.test.ts [hunk 3]
```diff
@@ -37,12 +37,12 @@ describe('auth login', () => {
...
     await expect(command.run()).rejects.toMatchInlineSnapshot('[Error: EEXIT: 2]');
 
-    expect(readFile).toHaveBeenCalledWith(credentialsPath, 'utf-8');
+    expect(readFile).toHaveBeenCalledWith(credentialsFilePath, 'utf-8');
     expect(promptsMock).toHaveBeenCalledOnce();
     expect(promptsMock.mock.calls[0]).toMatchInlineSnapshot(`
       [
         {
-          "message": "Authentication is already configured, do you want to overwrite it?",
+          "message": "Authentication is already configured for default profile, do you want to overwrite it?",
```
Comment (SferaDev) [inline cli/src/commands/auth/login.test.ts:45]:
```suggestion
          "message": "Authentication is already configured for the default profile, do you want to overwrite it?",
```

**Pull request merged**